src/client/remoteinputhandler.h

```diff
@@ -0,0 +1,155 @@
+#include "gui/mainmenumanager.h"
+#include "inputhandler.h"
+#include "remoteclient.capnp.h"
+#include <capnp/message.h>
+#include <capnp/serialize-packed.h>
+#include <zmqpp/zmqpp.hpp>
+
+class RemoteInputHandler : public InputHandler {
+public:
+  RemoteInputHandler(const std::string &endpoint,
+                     RenderingEngine *rendering_engine,
+                     MyEventReceiver *receiver)
+      : m_rendering_engine(rendering_engine), m_context(),
+        m_socket(m_context, zmqpp::socket_type::reply), m_receiver(receiver) {
+    infostream << "RemoteInputHandler: Binding to " << endpoint << '\n';
+    m_socket.bind(endpoint);
+    // wait for client
+    zmqpp::message message;
+    m_socket.receive(message);
+    std::string data;
```

Comment:
I'd factor out the code for reading an action from the socket since it's duplicated here and in the .cpp file.

And probably move the implementation of this constructor to the .cpp file.